Source/WebCore/platform/text/TextFlags.h

```diff
@@ -163,7 +166,64 @@ enum class FontVariantNumericFraction : uint8_t {
 
 enum class FontVariantNumericOrdinal : bool { Normal, Yes };
 enum class FontVariantNumericSlashedZero : bool { Normal, Yes };
-enum class FontVariantAlternates : bool { Normal, HistoricalForms };
+
+struct FontVariantAlternatesNormal {
+    bool operator==(const FontVariantAlternatesNormal&) const = default;
+};
+
+struct FontVariantAlternatesValues {
+    bool operator==(const FontVariantAlternatesValues&) const = default;
+    std::optional<String> stylistic;
+    std::optional<String> styleset;
+    std::optional<String> characterVariant;
+    std::optional<String> swash;
+    std::optional<String> ornaments;
+    std::optional<String> annotation;
+    bool historicalForms = false;
+    friend void add(Hasher&, const FontVariantAlternatesValues&);
+};
+struct FontVariantAlternates {
```

Comment:
If the members are private, it should be a class, and the members' names should start with `m_`.